torch_xla/csrc/xla_lower_util.cpp

```diff
@@ -135,4 +135,20 @@ xla::XlaOp BuildDropout(const xla::XlaOp& input, float probability) {
   return input * mask;
 }
 
+std::vector<xla::XlaOp> CreateBroadcastTensors(
+    tensorflow::gtl::ArraySlice<const xla::XlaOp> operands) {
+  XLA_CHECK(!operands.empty()) << "broadcast_tensors cannot take an empty list";
+  xla::Shape result_shape = XlaHelpers::ShapeOfXlaOp(operands.front());
+  for (const xla::XlaOp operand : operands) {
+    result_shape = XlaHelpers::GetPromotedShape(
+        result_shape, XlaHelpers::ShapeOfXlaOp(operand));
```

Comment:
Getting a Shape from the builder requires rebuilding it from proto, and trigger all validations.
Since you are using them in the next loop, caching them might not be a bad idea.

Reply:
Done.